File: ReservationFrontend/src/models/Reservation.ts
import { DateTime } from "./DateTime";

export interface Reservation {
    id: Number,
    startTime: DateTime,
    endTime: DateTime,
    name: String
  }
export interface ReservationDTO {
    name: String,
    startTime: String;
    endTime: String;
    password:String;
  }

File: ReservationBackend/src/main/java/com/example/ReservationBackend/Models/Reservation.java
package com.example.ReservationBackend.Models;


public class Reservation {
    public int id;
    public DateTime startTime;
    public DateTime endTime;
    public String name;

    public Reservation(int index,ReservationDTO reservationDTO) {
        this.id=index;
        this.startTime=new DateTime(reservationDTO.startTime);
        this.endTime=new DateTime(reservationDTO.endTime);
        this.name = reservationDTO.name;
    }

}
